src/ai/backend/test/templates/session/utils.py

```diff
@@ -0,0 +1,136 @@
+import asyncio
+from typing import Awaitable, Callable
+from uuid import UUID
+
+from ai.backend.client.func.session import ComputeSession
+from ai.backend.client.session import AsyncSession
+from ai.backend.common.json import load_json
+
+
+async def verify_batch_session(
+    client_session: AsyncSession,
+    session_name: str,
+    timeout: float,
+    session_factory: Callable[[], Awaitable["ComputeSession"]],
+) -> UUID:
+    EXPECTED_EVENTS: set[str] = {
+        "session_enqueued",
+        "session_scheduled",
+        "kernel_preparing",
+        "kernel_creating",
+        "kernel_started",
+        "session_started",
+        "session_terminating",
+        "session_terminated",
+        "kernel_terminating",
+        "kernel_terminated",
+        "session_success",
+    }
+
+    collected_events: set[str] = set()
+
+    async def collect_events() -> None:
+        async with client_session.ComputeSession(session_name).listen_events() as events:
+            async for ev in events:
+                collected_events.add(ev.event)
+                if ev.event == "session_failure" or ev.event == "session_cancelled":
+                    raise RuntimeError(f"BatchSession failed with event: {ev.event}")
+                if collected_events == EXPECTED_EVENTS:
+                    break
```

Comment:
When thinking from a user scenario perspective, instead of subscribing to receive all events, wouldn't it be sufficient to subscribe to receive only the events of the desired final state?